modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultConstantFoldingOptimizationPhase.java

```diff
@@ -621,22 +621,19 @@ public void visitStringConcatenation(StringConcatenationNode irStringConcatenati
             int j = i;
             irRightNode.visit(this, (e) -> irStringConcatenationNode.getArgumentNodes().set(j + 1, e));
 
-            if (irLeftNode instanceof ConstantNode && irRightNode instanceof ConstantNode) {
-                ConstantNode irConstantNode = (ConstantNode) irLeftNode;
+            if (irLeftNode instanceof ConstantNode irConstantNode && irRightNode instanceof ConstantNode) {
```

Comment:
This seems slightly inconsistently used here, in that the cast to `ConstantNode` seems unnecessary. I would prefer to either cast/pattern match both left and right nodes or none of them.

Reply:
Yeah, I didn't particularly like this block of code changes either. Seems too compact and the code too similar between the conditional branches to make it easy to read. I've reverted the changes. Even if the instanceof pattern matching is not used, the readability of the code is slightly better this way.